File: 프로그래머스/0/120826. 특정 문자 제거하기/특정 문자 제거하기.java
class Solution {
    public String solution(String my_string, String letter) {
        String answer = "";
        for(int i=0;i<my_string.length();i++){
            // letter 과 i번째 인덱스 문자가 같으면
            if(my_string.charAt(i)==letter.charAt(0)){
                continue; // 다음 문자로 넘어간다
            }
            // 다르면
            else{
                answer += my_string.charAt(i); // answer에 해당 문자 추가
            }
        }
        return answer;
    }
}

File: 프로그래머스/0/120826. 특정 문자 제거하기/특정 문자 제거하기.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// 파라미터로 주어지는 문자열은 const로 주어집니다. 변경하려면 문자열을 복사해서 사용하세요.
char* solution(const char* my_string, const char* letter) {
    // return 값은 malloc 등 동적 할당을 사용해주세요. 할당 길이는 상황에 맞게 변경해주세요.
    int len = strlen(my_string);
    char target = letter[0]; // letter의 첫 번째 문자로 비교
    char* answer = (char*)malloc(len + 1); // 널 종료문자 포함 공간 할당
    int index = 0; // answer의 현재 위치를 추적
    
    for (int i = 0; i < len; i++) {
        if (my_string[i] != target) {
            answer[index++] = my_string[i]; // 조건에 맞으면 문자 추가
        }
    }
    answer[index] = '\0'; // 문자열 종료
    
    return answer;
}